cli/src/commands/import/csv.ts

```diff
@@ -69,100 +67,195 @@ export default class ImportCSV extends BaseCommand<typeof ImportCSV> {
-        error = true;
-        this.warn(
-          `Column ${type.columnName} exists with type ${type.schemaType} but a type of ${type.castedType} would be needed.`
+    let importSuccessCount = 0;
+    const errors: string[] = [];
+    let progress = 0;
+    const fileStream = await getFileStream();
+    await xata.import.parseCsvStreamBatches({
+      fileStream: fileStream,
+      fileSizeBytes: await getFileSizeBytes(file),
+      parserOptions: { ...csvOptions, columns },
+      batchRowCount: batchSize,
+      onBatch: async (parseResults, meta) => {
+        if (!parseResults.success) {
+          throw new Error('Failed to parse CSV file');
+        }
+        const dbBranchName = `${database}:${branch}`;
+        const importResult = await xata.import.importBatch(
+          // @ts-ignore
+          { dbBranchName: dbBranchName, region, workspace: workspace, database },
+          { columns: parseResults.columns, table, batch: parseResults }
+        );
+        importSuccessCount += importResult.successful.results.length;
+        if (importResult.errors) {
+          const formattedErrors = importResult.errors.map(
+            (error) => `${error.error}. Record: ${JSON.stringify(error.row)}`
+          );
+          const errorsToLog = formattedErrors.slice(0, Math.abs(ERROR_CONSOLE_LOG_LIMIT - errors.length));
+          for (const error of errorsToLog) {
+            this.logToStderr(`Import Error: ${error}`);
+          }
+          errors.push(...formattedErrors);
+        }
+        progress = Math.max(progress, meta.estimatedProgress);
+        this.info(
+          `${importSuccessCount} rows successfully imported ${errors.length} errors. ${Math.ceil(
+            progress * 100
+          )}% complete`
         );
       }
     });
-    if (error) {
-      return process.exit(1);
+    if (errors.length > 0) {
+      await writeFile(ERROR_LOG_FILE, errors.join('\n'), 'utf8');
+      this.log(`Import errors written to ${ERROR_LOG_FILE}`);
     }
+    fileStream.close();
+    this.success('Completed');
+    process.exit(0);
+  }
 
-    if (compare.missingTable) {
-      if (!create) {
-        const response = await this.prompt({
-          type: 'confirm',
-          name: 'confirm',
-          message: `Table ${table} does not exist. Do you want to create it?`,
-          initial: false
-        });
-        if (!response.confirm) return false;
-      }
-    } else if (compare.missingColumns.length > 0) {
-      if (!create) {
-        const response = await this.prompt({
+  async parseDatabase() {
+    const { flags } = await this.parseCommand();
+    return await this.getParsedDatabaseURLWithBranch(flags.db, flags.branch);
+  }
+
+  async migrateSchema({
+    table,
+    columns,
+    create
+  }: {
+    table: string;
+    columns: Column[];
+    create: boolean;
+  }): Promise<void> {
+    const xata = await this.getXataClient();
+    const { workspace, region, database, branch } = await this.parseDatabase();
```

Comment:
I was prompted 2x for the database I wanted, we should re-use the destination info from the root call of the command

<img width="573" alt="Screenshot 2023-07-07 at 09 33 53" src="https://github.com/xataio/client-ts/assets/2181866/5e0dfd22-2c13-4649-b548-6c8e973d2416">

Reply:
https://github.com/xataio/client-ts/pull/1051/commits/56ffba1931ee5cdc5b75989016d87fca312c8d47 ✅